### PR TITLE
P2: customize signup page

### PR DESCRIPTION
**Pull request by annemirasol**
#### Changes proposed in this Pull Request

* Customizes the signup page for P2 flows, as described in p9lV3a-2Lp-p2. This is part of the P2 Onboarding project.
* Refactor: move parts of the customized styles for `accept-invite` signup to `logged-out-form` so it can be reused by the `start/p2` flow

#### Testing instructions 

* Regression test: verify that non-P2 signup flows have the regular signup screen.
* Regression test: verify that accepting invites for non-P2 sites have the regular signup screen.
* Check that nothing looks broken for the following:
    * sign up screen when accepting P2 invites
    * sign up screen for the `/start/p2` flow 
* Check that you can successfully sign up via the `/start/p2` flow 

### Screens
1) Before expanding the email signup form

Screen Shot 2021-08-26 at 11 18 15 PM

2) After expanding the email signup form (at 80%)

Screen Shot 2021-08-26 at 11 18 27 PM

3) (Refactored so styles for email signup form are shared) The `accept-invite-logged-out` screen

Screen Shot 2021-08-26 at 10 21 33 PM

### CODE REVIEW

File: client/blocks/signup-form/p2.jsx
Comment (a8ci18n) [inline client/blocks/signup-form/p2.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **12** times:
`translate( 'or', { context: 'Custom Background: Choose an image from your computer - or - Choose from image library'} )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/p2-signup)


:information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `h4-lp-option-divider`
- `Shown between widget visibility conditions.`
- `Uploader: Drop files here - or - Select Files`

 Would it make sense to reuse one of the above?

**Comment by lezama** [comment]
Buttons on this screen look a little bit narrower for me: 
Screen Shot 2021-08-27 at 09 55 53

**Comment by annemirasol** [comment]
> Buttons on this screen look a little bit narrower for me:
> 

Fixed.

**Review by unDemian — CHANGES_REQUESTED** [review]
Code looks good! 🎉  I've successfully tested the following

✅ Regression test: verify that non-P2 signup flows have the regular signup screen.
✅ Regression test: verify that accepting invites for non-P2 sites have the regular signup screen.

Check that nothing looks broken for the following:
✅ sign up screen when accepting P2 invites
✅ sign up screen for the /start/p2 flow

Not sure if this was introduced by this PR but I think we should take a look at error handling during sign up.

❌ Check that you can successfully sign up via the /start/p2 flow
- From the accept invite url if I try to sign up and I encounter an error it is not displayed and the screen just feels stuck https://d.pr/i/294koB
- From `/start/p2` if I sign up and encounter an error I get redirected around with no indication of what happened https://d.pr/v/Eta5mT

**Comment by annemirasol** [comment]
Good catch, thanks!

> From the accept invite url if I try to sign up and I encounter an error it is not displayed and the screen just feels stuck https://d.pr/i/294koB

This will be fixed by #56200 and #56202

> From /start/p2 if I sign up and encounter an error I get redirected around with no indication of what happened https://d.pr/v/Eta5mT

Fixed in this PR.

File: client/blocks/signup-form/index.jsx
Comment (a8ci18n) [inline client/blocks/signup-form/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **55** times:
`translate( 'Your password cannot be the same as your username. Please pick a different password.' )` *ES Score: 7*

Comment (annemirasol) [reply client/blocks/signup-form/index.jsx]: Done.

**Review by unDemian — APPROVED** [review]
Works as expected! 💯

**Comment by evilluendas** [comment]
Looks fantastic! I just pushed a handful of small tweaks to make it more consistent with the mockups, and fixed a small bug in the current first step (a longer "Choose an address for your P2 _workspace_" label made the `.wordpress.com` suffix in the url field be misaligned on mobile).

Everything seems to look as expected, but I hope I haven't broken anything!

Screen Shot 2021-09-17 at 19 57 55

Screen Shot 2021-09-17 at 19 58 08

|How it looks now|How it'll look once merged|
|-|-|
|image|image|

File: client/my-sites/invites/p2/invite-accept-logged-out.jsx
Comment (a8ci18n) [inline client/my-sites/invites/p2/invite-accept-logged-out.jsx:14]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6637739

Thank you @annemirasol for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6637739).

**Pull request merged**
